### DISCUSSION TITLE
Reduce capacity after conversion from gltf to glb

### DISCUSSION THREAD

**Discussion by EthanKyleKim** [donmccurdy/glTF-Transform#1585] [Q&A]
I'm glad to see you.

I just worked on changing .gltf file to .glb.
To go into detail

I used toktx to convert .png into ktx2 format and then super-compressed it again using ktxsc

But after all this texturing, I converted gltf into .glb
It came out to 167MB. I want a smaller capacity than this

The capacity I want is about 30MB, but I'm not sure if it's this possible.

I was wondering how to reduce the size of the glb file as much as possible while maintaining good quality.
The resolution is 4096x4096. 
Should there be no problem if I change it from 4096 to 2048 or lower resolution?

Below is the value I used when I applied toktx and ktxsc

```toktx --uastc --zcmp 19```
```ktxsc --encode uastc --zcmp 19 -o```

And these are the texture files I'm using I converted from png to ktx2.
![image](https://github.com/user-attachments/assets/535f1690-2cf1-4dc3-973f-06acd7b8c1bb)

devtools
![image](https://github.com/user-attachments/assets/fd74e855-2e29-4d36-b4c6-2a4d7070b80b)

I need your knowledge!

**Comment by donmccurdy** [comment]
Hi @EthanKyleKim! It's not clear to me how you're using the glTF Transform library, or if you are?  

If your original textures are part of a .gltf/.glb file then that file would need to be updated to reference the new (.ktx2) textures after KTX2 compression, the 'toktx' CLI alone does not do that. In the gltf-transform CLI there's a built-in feature for UASTC compression that does this repackaging for you, see `gltf-transform uastc --help`. But this is still using the same KTX Software compression so I don't think your results will be much different, unless you also reduce resolution, see the help output for how to do that.

If you're still not able to reduce the size of the file, and if you can share the file, and I might be able to spot something by looking at it.
